components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicClientAuthenticationHandler.java

```diff
@@ -0,0 +1,185 @@
+                        } else {
+                            authenticationResult.setAuthenticationStatus(AuthenticationStatus.FAILED);
+                        }
+                    } catch (IdentityOAuthAdminException e) {
+                        String errorMessage = "Error while authenticating application with client ID: " + clientId;
+                        log.error(errorMessage, e);
+                        throw new AuthenticationFailException(errorMessage, e);
+                    } catch (InvalidOAuthClientException | IdentityOAuth2Exception e) {
+                        String errorMessage = "Invalid client: " + clientId;
+                        log.error(errorMessage, e);
+                        throw new AuthenticationFailException(errorMessage, e);
+                    }
+                } else {
+                    String errorMessage = "Error occurred while trying to authenticate. The OAuth application credentials "
+                            + "are not defined correctly.";
+                    log.error(errorMessage);
+                    throw new AuthenticationFailException(errorMessage);
+                }
+            } else {
+                String errorMessage = "Error occurred while trying to authenticate. The " + HttpHeaders.AUTHORIZATION +
+                        " header values are not defined correctly.";
+                log.error(errorMessage);
+                throw new AuthenticationFailException(errorMessage);
+            }
+        } else {
+            String errorMessage = "Authorization Header is null.";
+            log.error(errorMessage);
+            throw new AuthenticationFailException(errorMessage);
+        }
+        return authenticationResult;
+    }
+
+    private OAuthAppDO getOAuthApplication(String consumerKey) throws InvalidOAuthClientException,
+            IdentityOAuth2Exception {
+
+        OAuthAppDO authAppDO = OAuth2Util.getAppInformationByClientId(consumerKey);
+        String appState = authAppDO.getState();
+        if (StringUtils.isEmpty(appState)) {
+            if (log.isDebugEnabled()) {
+                log.debug("A valid OAuth client could not be found for client_id: " + consumerKey);
+            }
+            throw new InvalidOAuthClientException("A valid OAuth client not found for client_id: " + consumerKey);
+        }
+
+        if (!APP_STATE_ACTIVE.equalsIgnoreCase(appState)) {
+            if (log.isDebugEnabled()) {
+                log.debug("App is not in active state in client ID: " + consumerKey + ". App state is:" + appState);
+            }
+            throw new InvalidOAuthClientException("Oauth application is not in active state");
+        }
+        return authAppDO;
+    }
+
+    private void validateRequestTenantDomain(String tenantDomainOfApp) throws InvalidOAuthClientException {
+
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
```

Comment:
what happens if isTenantQualifiedUrlsEnabled = flase?

Reply:
REST API URLs should be tenant qualified anyway. SO. we don't need to check this check.

Reply:
Client ID will not be validated against the tenant domain.

Reply:
addressed with commit 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